src/typeahead/typeahead.ts

```diff
@@ -367,7 +367,7 @@ export class NgbTypeahead implements ControlValueAccessor,
     this._renderer.setProperty(this._elementRef.nativeElement, 'value', toString(value));
   }
 
-  private _subscribeToUserInput(userInput$: Observable<any[]>): Subscription {
+  private _subscribeToUserInput(userInput$: Observable<ReadonlyArray<any>>): Subscription {
```

Comment:
Let's drop this as a non-public API

Reply:
This one will need to stay as read-only (but I'll change it to `readonly any[]`) because the value comes from result of `ngbTypeahead`.

```ts
const results$ = inputValues$.pipe(this.ngbTypeahead);
const userInput$ = this._resubscribeTypeahead.pipe(switchMap(() => results$));
this._subscription = this._subscribeToUserInput(userInput$);
```
```
Argument of type 'Observable<readonly any[]>' is not assignable to parameter of type 'Observable<any[]>'.
  The type 'readonly any[]' is 'readonly' and cannot be assigned to the mutable type 'any[]'.ts(2345)
```